common/src/models/deployment.rs

```diff
@@ -7,12 +7,21 @@ use comfy_table::{
 };
 use crossterm::style::Stylize;
 use serde::{Deserialize, Serialize};
+use utoipa::IntoParams;
 #[cfg(feature = "openapi")]
 use utoipa::ToSchema;
 use uuid::Uuid;
 
 use crate::deployment::State;
 
+#[derive(Debug, Clone, Copy, Deserialize, IntoParams)]
+pub struct PaginationDetails {
```

Comment:
Late review comment: I am not sure what this looks like from the perspective of a client who wants to find how many pages to display in a browser page, listing the deployments (same for the projects). It will not be able to display a certain number of pages because it's not knowing how many it can display. I think this is a separate API that has to be done by the client. I also think that another PR can address this. What do you guys think?

Reply:
I just had a sync with @Kazy and we can omit resolving the issue I've described because the functionality can be still used like it is. The reason is that the pagination API returns an empty response list for an out-of-bounds page. This means that the web UIs who want to display page numbers will be able to understand where to stop. Assuming the UIs display a limited interval of pages, it will need to do requests for for all of them and if a request for a page returns an empty list, the client knows it reached the end. I will resolve this since it's a non-issue for now and will be relevant only for cases where the clients prefer to do some kind of lazy fetching which will require a pre-fetching of the projects count or this total count added with each page response.